improver/blending/spatial_weights.py

```diff
@@ -289,15 +150,150 @@ def create_template_slice(cube_to_collapse, blend_coord):
+        normalised_weights = iris.cube.CubeList()
+        for weights_slice in weights.slices_over(self.blend_coord):
+            normalised_data = np.where(
+                weights_sum > 0, np.divide(weights_slice.data, weights_sum), 0
+            )
+            normalised_weights.append(
+                weights_slice.copy(data=normalised_data.astype(FLOAT_DTYPE))
+            )
+        weights = normalised_weights.merge_cube()
+        return weights
+
+    def _rescale_masked_weights(self, weights):
+        """Apply fuzzy smoothing to weights at the edge of masked areas
+
+        Args:
+            weights (iris.cube.Cube):
+                Pre-normalised weights where the weights of masked data points
+                have been set to 0
+
+        Returns:
+            (tuple): tuple containing:
+                **weights** (iris.cube.Cube):
+                    Weights where MASKED slices have been rescaled, but UNMASKED
+                    slices have not
+                **rescaled** (iris.cube.Cube):
+                    Binary (0/1) map showing which weights have been rescaled
+
+        """
+        is_rescaled = iris.cube.CubeList()
+        rescaled_weights = iris.cube.CubeList()
+        for weights_slice in weights.slices_over(self.blend_coord):
+            weights_nonzero = np.where(weights_slice.data > 0, 1, 0)
+            if np.all(weights_nonzero == 1):
+                # if there are no masked points in this slice, keep current weights
+                # and mark as unchanged (not rescaled)
+                rescaled_weights.append(weights_slice)
+                is_rescaled.append(
+                    weights_slice.copy(data=np.zeros(weights_slice.shape, dtype=int))
+                )
+            else:
+                weights_orig = weights_slice.data.copy()
+
+                # calculate the distance to the nearest invalid point, in grid squares,
+                # for each point on the grid
+                distance = distance_transform_edt(weights_nonzero)
+
+                # calculate a 0-1 scaling factor based on the distance from the
+                # nearest invalid data point, which scales between 1 at the fuzzy length
+                # towards 0 for points closest to the edge of the mask
+                fuzzy_factor = rescale(
+                    distance, data_range=[0.0, self.fuzzy_length], clip=True
+                )
+
+                # multiply existing weights by fuzzy scaling factor
+                rescaled_weights_data = np.multiply(
+                    weights_slice.data, fuzzy_factor
+                ).astype(FLOAT_DTYPE)
+                rescaled_weights.append(weights_slice.copy(data=rescaled_weights_data))
+
+                # identify spatial points where weights have been rescaled
+                is_rescaled_data = np.where(
+                    rescaled_weights_data != weights_orig, 1, 0
+                ).astype(int)
```

Comment:
Same as above, I think this could be `bool`

Reply:
Done.